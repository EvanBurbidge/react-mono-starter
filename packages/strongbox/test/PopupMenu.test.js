import React from 'react';
import {PopupMenu} from '../build/rrpm.module';
import renderer from 'react-test-renderer';

describe('PopupMenu', () => {
  const component = renderer.create(
    <PopupMenu />
  );

  let tree;

  it('should render correctly', () => {
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Hover button
  it('should respond to mouse hover', () => {
    tree.children[0].props.onMouseOver();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Unhover button
  it('should respond to mouse unhover', () => {
    tree.children[0].props.onMouseOut();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
