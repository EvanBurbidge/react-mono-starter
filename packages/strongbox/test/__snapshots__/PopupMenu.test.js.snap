// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupMenu should render correctly 1`] = `
<div
  className="PopupMenu_PopupMenu__8TfA4"
>
  <div
    className="PopupMenu_button__20m_Y"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  />
  <div
    className="PopupMenu_popover__3hd_Z"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "height": "auto",
        "left": "calc(-100px + 50%)",
        "visibility": "hidden",
        "width": "calc(200px - 10px)",
      }
    }
  />
</div>
`;

exports[`PopupMenu should respond to mouse hover 1`] = `
<div
  className="PopupMenu_PopupMenu__8TfA4"
>
  <div
    className="PopupMenu_button__20m_Y active"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  />
  <div
    className="PopupMenu_popover__3hd_Z active"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "height": "auto",
        "left": "calc(-100px + 50%)",
        "visibility": "",
        "width": "calc(200px - 10px)",
      }
    }
  />
</div>
`;

exports[`PopupMenu should respond to mouse unhover 1`] = `
<div
  className="PopupMenu_PopupMenu__8TfA4"
>
  <div
    className="PopupMenu_button__20m_Y"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  />
  <div
    className="PopupMenu_popover__3hd_Z"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "height": "auto",
        "left": "calc(-100px + 50%)",
        "visibility": "",
        "width": "calc(200px - 10px)",
      }
    }
  />
</div>
`;
